### PR TITLE
enh(typescript): highlight obvious user defined type names (`interface`, etc)

### PR DESCRIPTION
**Pull request by MohamedAli00949**
I have fixed highlighting params at JS and these params types such as (interfaces, types, etc) at TS.

The related issue is #3269

### Changes

My changes are concat the params regex contains with class reference regex and attribute regex to highlight props keys and types.
While I working on this issue, I noted that the params didn't highlight if it separated with space before and solved that also.
### Checklist
- [x] Fix params highlighting at js functions.
- [x] Fix params types (interfaces, types, etc).
- [x] Added markup tests to test the changes.

### CODE REVIEW

**Comment by MohamedAli00949** [comment]
Hi @joshgoebel, this is my suggestion and I hope to review it.

**Comment by MohamedAli00949** [comment]
@joshgoebel, is there any opportunity to review these changes?
Thanks in advance

File: src/languages/typescript.js [hunk 1]
```diff
@@ -101,14 +101,41 @@ export default function(hljs) {
...
+        const PROPERTY_TYPE = [
+          CLASS_REFERENCE,
+          ATTRIBUTE_HIGHLIGHT
+        ];
+
+        if (params) {
+          params.contains = params.contains.concat(PROPERTY_TYPE)
+        }
+
+        const returnedFuncParams = contains.find(cc => cc.scope === 'function' || cc.className === 'function')?.contains?.find(cc => cc.scope === 'params' || cc.className === 'params').variants[2];
```
Comment (joshgoebel) [inline src/languages/typescript.js:131]: The `function` scope is deprecated and will be removed in the future...

File: src/languages/typescript.js [hunk 2]
```diff
@@ -101,14 +101,41 @@ export default function(hljs) {
...
       'mts',
       'cts'
-    ]
+    ],
+    contains: tsLanguage.contains.map(c => {
+      const { contains } = c;
+      if (
+        contains
+      ) {
+        const params = contains.find(cc => cc.scope === 'params' || cc.className === 'params');
```
Comment (joshgoebel) [inline src/languages/typescript.js:120]: Do you really need params here or just access to params_contains?  See the already exported `tsLanguage.exports.PARAMS_CONTAINS` instead.

File: src/languages/typescript.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ import * as ECMAScript from "./lib/ecmascript.js";
 import javascript from "./javascript.js";
 
 /** @type LanguageFn */
-export default function(hljs) {
+export default function (hljs) {
```
Comment (joshgoebel) [inline src/languages/typescript.js:14]: Please remove the linting/spacing only changes.

File: src/languages/javascript.js [hunk 4]
```diff
@@ -256,15 +256,15 @@ export default function(hljs) {
...
+      begin: /(\s*)\(/,
       end: /\)/,
       keywords: KEYWORDS,
       contains: ["self"].concat(SUBST_AND_COMMENTS)
     }
   ]);
   const PARAMS = {
     className: 'params',
-    begin: /\(/,
+    begin: /(\s*)\(/, // to match the parms with
```
Comment (joshgoebel) [inline src/languages/javascript.js:267]:
```suggestion
    // convert this to negative lookbehind in v12
    begin: /(\s*)\(/, // to match the parms with 
```

File: src/languages/typescript.js [hunk 5]
```diff
@@ -87,6 +87,13 @@ export default function(hljs) {
   Object.assign(tsLanguage.keywords, KEYWORDS);
 
   tsLanguage.exports.PARAMS_CONTAINS.push(DECORATOR);
+
+  // highlight the function params
+  const ATTRIBUTE_HIGHLIGHT = tsLanguage.contains.find(c => c.className === "attr");
+  tsLanguage.exports.PARAMS_CONTAINS.push([
+    tsLanguage.exports.CLASS_REFERENCE, // class reference for highlighting the params types
+    ATTRIBUTE_HIGHLIGHT, // highlight the params key
+  ]);
```
Comment (joshgoebel) [inline src/languages/typescript.js:96]: Much simpler!

File: src/languages/javascript.js (hunk 4 above)
Comment (MohamedAli00949) [reply src/languages/javascript.js:267]: do you mean to remove this change or add this `// convert this to negative lookbehind in v12` comment above the begin?
I am sorry, i can't understand you.

File: src/languages/typescript.js (hunk 5 above)
Comment (MohamedAli00949) [reply src/languages/typescript.js:96]: That's right

File: src/languages/javascript.js (hunk 4 above)
Comment (joshgoebel) [reply src/languages/javascript.js:267]: Just add the comment.  I wouldn't have suggested the change otherwise.

Comment (MohamedAli00949) [reply src/languages/javascript.js:267]: Ok, that's nice to add this comment.
Thank you

**Comment by joshgoebel** [comment]
Now we just need a changelog entry I think.

**Comment by joshgoebel** [comment]
Looks like markup tests aren't all passing?

**Comment by MohamedAli00949** [comment]
> Looks like markup tests aren't all passing?

I tested them and all have been passed before creating the PR and after modification at Typescript and all passed.

If you have found any failed test comment it to fix it.
Thanks in advance

**Comment by joshgoebel** [comment]
See the failure here: https://github.com/highlightjs/highlight.js/actions/runs/6872592218/job/18691291704?pr=3903

Are you saying they all pass for you locally? Is your working tree clean?

**Comment by MohamedAli00949** [comment]
> See the failure here: https://github.com/highlightjs/highlight.js/actions/runs/6872592218/job/18691291704?pr=3903
> 
> Are you saying they all pass for you locally? Is your working tree clean?

You are right, I am sorry.

This is happening because I am running an `npm run test` command to test all test types.
But the markup tests have passed.

Screenshot 2023-11-16 032743

And the result of `npm run test-markup` is the result at the above image.

Screenshot 2023-11-16 032521


I will commit changing at `declares.expect.txt` and `jsx.expect.txt` now.

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**